projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -216,22 +219,24 @@ private InboundFrame(FrameType type, int channel, ReadOnlyMemory<byte> payload,
             _rentedArray = rentedArray;
         }
 
-        private static void ProcessProtocolHeader(Stream reader, ReadOnlySpan<byte> frameHeader)
+        private static void ProcessProtocolHeader(ReadOnlySequence<byte> buffer)
         {
             try
             {
-                if (frameHeader[0] != 'M' || frameHeader[1] != 'Q' || frameHeader[2] != 'P')
+                if (buffer.Length < 8)
                 {
-                    throw new MalformedFrameException("Invalid AMQP protocol header from server");
+                    throw new EndOfStreamException();
                 }
 
-                int serverMinor = reader.ReadByte();
-                if (serverMinor == -1)
+                Span<byte> tempSpan = stackalloc byte[8];
```

Comment:
I might be missing something. Why is the copy needed here? The buffer should be reachable for the check below and not go out of scope.